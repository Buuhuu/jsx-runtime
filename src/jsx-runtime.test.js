/*
 * Copyright 2022 Dirk Rudolph
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { getByText } from "@testing-library/dom"

describe('hlx jsx runtime', () => {

    it('renders simple dom nodes', () => {
        const nodes = (
            <div>
                <h1>Hello World</h1>
                <p>This is a unordered list of numbers:</p>
                <ul>
                    {[1, 2, 3].map(number => <li>{number}</li>)}
                </ul>
                <a href="https://letmegooglethat.com/?q=what+is+jsx">Find out more</a>
            </div>
        )

        expect(nodes).toMatchSnapshot()
    })

    it('renders dom nodes with styles', () => {
        const nodes =
            <div>
                <p style={{ color: 'red', fontSize: 'large' }}>Hello World</p>
            </div>


        expect(nodes).toMatchSnapshot()
    })

    it('renders a fragment with mutliple children', () => {
        const nodes =
            <>
                <p>First Paragraph</p>
                <p>Second Paragraph</p>
                <p>Third Paragraph</p>
            </>


        expect(nodes).toMatchSnapshot()
    })

    it('renders a fragment with a single child', () => {
        const nodes = <>Text Content</>

        expect(nodes).toMatchSnapshot()
    })

    it('renders an empty fragment', () => {
        const nodes = <></>

        expect(nodes).toMatchSnapshot()
    })

    it('renders components', () => {
        const Component = ({ text }) => <p>{text}</p> // eslint-disable-line no-unused-vars
        const Container = ({ children }) => <p>{children}</p> // eslint-disable-line no-unused-vars
        const nodes = (
            <div>
                <Component text="foobar" />
                <Container>
                    Foo Bar
                    <em>Foo Bar</em>
                </Container>
            </div>
        )

        expect(nodes).toMatchSnapshot()
    })

    it('renders listeners', () => {
        let clicked = false
        const nodes = <button onclick={() => clicked = true}>Button</button>
        const button = getByText(nodes, 'Button')
        
        button.click()

        expect(clicked).toBeTruthy();
    })
})
