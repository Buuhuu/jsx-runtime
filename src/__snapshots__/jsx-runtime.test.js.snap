// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hlx jsx runtime renders a fragment with a single child 1`] = `
<DocumentFragment>
  Text Content
</DocumentFragment>
`;

exports[`hlx jsx runtime renders a fragment with mutliple children 1`] = `
<DocumentFragment>
  <p>
    First Paragraph
  </p>
  <p>
    Second Paragraph
  </p>
  <p>
    Third Paragraph
  </p>
</DocumentFragment>
`;

exports[`hlx jsx runtime renders an empty fragment 1`] = `<DocumentFragment />`;

exports[`hlx jsx runtime renders components 1`] = `
<div>
  <p>
    foobar
  </p>
  <p>
    Foo Bar
    <em>
      Foo Bar
    </em>
  </p>
</div>
`;

exports[`hlx jsx runtime renders dom nodes with styles 1`] = `
<div>
  <p
    style="color: red; font-size: large;"
  >
    Hello World
  </p>
</div>
`;

exports[`hlx jsx runtime renders simple dom nodes 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <p>
    This is a unordered list of numbers:
  </p>
  <ul>
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
  </ul>
  <a
    href="https://letmegooglethat.com/?q=what+is+jsx"
  >
    Find out more
  </a>
</div>
`;
